lib/parser.js

```diff
@@ -487,7 +488,11 @@ Parser.prototype._rebaseReferenceForStaticIncludes = function (pageData, element
 };
 
 function calculateBoilerplateFilePath(filePath) {
-  return path.resolve(path.join(process.cwd(), BOILERPLATE_DIRECTORY_NAME), path.basename(filePath));
+  let possibleBases = Object.keys(Parser.baseUrlMap).filter(base => pathIsInside(filePath, base));
+  let longest = possibleBases.sort(function (a, b) {
+    return b.length - a.length;
+  })[0];
```

Comment:
Seems related: `baseUrlFromRoot` in [Page.js#L35-L50](https://github.com/MarkBind/markbind-cli/blob/1ff54e85f239778aed8c3ba4567759cc8f3f3fca/lib/Page.js#L35-L50)

Reply:
Is that the same as https://github.com/MarkBind/markbind/blob/master/lib/parser.js#L483?

Reply:
Similar, but not the same. They should probably be renamed:
- markbind's parser.js: `calculateNewBaseUrl` → `calculateNewBaseUrlMap`
- markbind-cli's Page.js: `baseUrlFromRoot` → `calculateNewBaseUrl`

Not sure if the implementation can be improved, or if your algorithm is a comparable alternative.

Reply:
Edited to use `calculateNewBaseUrl` from `parser.js` and it seems to work.
But there might be a simpler way to implement both functions